ql/src/java/org/apache/hadoop/hive/ql/ddl/table/storage/compact/AlterTableCompactOperation.java

```diff
@@ -133,7 +138,7 @@ private List<Partition> getPartitions(Table table, AlterTableCompactDesc desc, D
       }
     } else {
       Map<String, String> partitionSpec = desc.getPartitionSpec();
-      partitions = context.getDb().getPartitions(table, partitionSpec);
+      partitions.addAll(PartitionUtils.getPartitions(context.getDb(), table, partitionSpec, false));
```

Comment:
you should amend `Hive.getPartitions`, similar to `Hive.getPartitionsByExpr`

Reply:
Done